In [9]:
import pandas as pd
import numpy as np
import math

1)Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [28]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
zp

array([ 35,  45, 190, 200,  40,  70,  54, 150, 120, 110])

In [29]:
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
ks

array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [30]:
cov=np.mean(ks*zp)-np.mean(ks)*np.mean(zp)
cov

9157.839999999997

In [31]:
np.cov(ks,zp,ddof=0)

array([[30468.89,  9157.84],
       [ 9157.84,  3494.64]])

In [32]:
np.std(ks)

174.55340157098058

In [33]:
np.std(zp)

59.115480206118605

In [36]:
9157.8399/(174.5534*59.1154)

0.8874912944919768

In [37]:
np.corrcoef(ks,zp)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

2)Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [46]:
iq= [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
iq

[131, 125, 115, 122, 131, 115, 107, 99, 125, 111]

In [59]:
x=np.mean(iq)
x

118.1

In [55]:
S= np.std(iq, ddof=1)
S

10.54566788359614

In [56]:
t=2.262

In [57]:
T1=x-t*S/math.sqrt(10)
T1

110.55660776308164

In [58]:
T1=x+t*S/math.sqrt(10)
T1

125.64339223691834

110.55 < a < 125.64

3)Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [62]:
T1=174.2-1.96*5/math.sqrt(27)
T1

172.31398912064722

In [64]:
T2=174.2+1.96*5/math.sqrt(27)
T2

176.08601087935276

T1 <  𝛼  < T2

172.31 < a < 176.08